hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/AbstractNegatableSqmPredicate.java

```diff
@@ -33,10 +33,16 @@ public void negate() {
 		this.negated = !this.negated;
 	}
 
+	protected abstract SqmNegatablePredicate createNegatedNode();
+
 	@Override
 	public SqmNegatablePredicate not() {
 		// in certain cases JPA required that this always return
-		// a new instance.  we may need to allow for that here (compliance?)
+		// a new instance.
+		if ( nodeBuilder().isJpaQueryComplianceEnabled() ) {
+			return createNegatedNode();
```

Comment:
IMO this should be the default. How else would you spell something like this if we alter the original predicate negation flag on a `not` call?

`predicate OR not(predicate) AND ...`

Reply:
Possibly.  See above.  If it were up to me, and we wanted to create new nodes for each negation, I'd simply wrap them all in `SqmNegatedPredicate`.  But like above, that messed with `#isNegated`